pytorch_lightning/trainer/trainer.py

```diff
@@ -449,17 +449,52 @@ def default_attributes(cls):
         return args
 
     @classmethod
-    def add_argparse_args(cls, parent_parser: ArgumentParser) -> ArgumentParser:
-        """Extend existing argparse by default `Trainer` attributes."""
-        parser = ArgumentParser(parents=[parent_parser], add_help=False)
+    def _get_argparse_args_and_types(cls) -> Generator[Tuple[str, Any, Any], None, None]:
+        r"""Generates arguments, which could extend custom `ArgumentParser` object.
 
-        trainer_default_params = Trainer.default_attributes()
+        Yields:
+            Tuple with 3 values: argument name, argument type and argument default value.
+        """
+        import inspect
```

Comment:
I would move the `inspect` to the beginning of the file as it is standard unless there is special reason